rosidl_generator_py/resource/_msg.py.em

```diff
@@ -182,45 +186,36 @@ class @(message.structure.type.name)(metaclass=Metaclass_@(message.structure.typ
 @[end for]@
     ]
 
-    _fields_and_field_types = {
+    SLOT_TYPES = (
 @[for member in message.structure.members]@
 @{
 type_ = member.type
 if isinstance(type_, NestedType):
     type_ = type_.basetype
 }@
-        '@(member.name)': '@
-@# the prefix for nested types
-@[  if isinstance(member.type, Sequence)]@
-sequence<@
+        @
+@[  if isinstance(member.type, Sequence) or isinstance(member.type, Array)]@
+@(member.type.__class__.__module__).@(member.type.__class__.__name__)(@
 @[  end if]@
 @# the typename of the non-nested type or the nested basetype
+@(type_.__class__.__module__).@(type_.__class__.__name__)(@
 @[  if isinstance(type_, BasicType)]@
-@(type_.type)@
+'@(type_.type)')@
 @[  elif isinstance(type_, BaseString)]@
-@
-@[    if isinstance(type_, WString)]@
-w@
-@[    end if]@
-string@
-@[    if type_.maximum_size is not None]@
-<@(type_.maximum_size)>@
-@[    end if]@
+'@(str(type_.maximum_size))')@
```

Comment:
Why should the integer size be quoted?

Reply:
It shouldn't :+1: ; looks like a remnant from previous testings, removed in https://github.com/ros2/rosidl_python/pull/33/commits/808f12ff0a021bbc17272535ac84e0ecd4db757c. I also removed the string conversion as writing `None` explicitly is not needed either